.cirrus.yml

```diff
@@ -639,6 +639,15 @@ local_integration_test_task: &local_integration_test_task
     alias: local_integration_test
     # Docs: ./contrib/cirrus/CIModes.md
     only_if: *not_tag_branch_build_docs_machine
+    # skip: - when PR (we never want to skip on nightly tests)
+    #       - when no danger files are changed
+    #       - when no int test code is changed
+    #       - NOT (source code is changed AND not only test files)
+    skip: &skip_int_test >-
+        $CIRRUS_PR != '' &&
+        !changesInclude('.cirrus.yml', 'Makefile', 'contrib/cirrus/**', 'vendor/**', 'hack/**') &&
```

Comment:
Sure would be nice to refactor this into `&skip_common`, but there does not seem to be any way to do that in YAML. I've given up after much too much wasted time. However, WDYT of a sanity check such as this?
```diff
diff --git a/contrib/cirrus/cirrus_yaml_test.py b/contrib/cirrus/cirrus_yaml_test.py
index f6de553c2..d93da811b 100755
--- a/contrib/cirrus/cirrus_yaml_test.py
+++ b/contrib/cirrus/cirrus_yaml_test.py
@@ -60,6 +60,18 @@ class TestDependsOn(TestCaseBase):
                 msg=('No success aggregation task depends_on "{0}"'.format(task_name))
                 self.assertIn(task_name, success_deps, msg=msg)
 
+    def test_skips(self):
+        """2024-06 PR#23030: ugly but necessary duplication in skip conditions. Prevent typos or unwanted changes."""
+        beginning = "$CIRRUS_PR != '' && !changesInclude('.cirrus.yml', 'Makefile', 'contrib/cirrus/**', 'vendor/**', 'hack/**') && "
+
+        for task_name in self.ALL_TASK_NAMES:
+            task = self.CIRRUS_YAML[task_name + '_task']
+            if 'skip' in task:
+                skip = task['skip']
+                if 'changesInclude' in skip:
+                    msg = ('{0}: invalid skip "{1}"'.format(task_name, skip))
+                    self.assertEqual(skip[:len(beginning)], beginning, msg=msg)
+
     def not_task(self):
         """Ensure no task is named 'task'"""
         self.assertNotIn('task', self.ALL_TASK_NAMES)
```

Reply:
good idea

Reply:
> but there does not seem to be any way to do that in YAML

Correct, scalar values cannot be merged/combined using anchors/aliases.  Only maps :(

One idea, and I haven't tested this:  Can you define patterns as global `env` values, then reference them inside `!changesInclude()`?  (I'm guessing not, but might be worth a try).

Reply:
Suggestion: Also add a `cirrus_yaml_test.py` test to make sure the skip conditions always include a `$CIRRUS_PR != ''`.  This prevents someone working on `.cirrus.yml` from accidentally enabling skips on the nightly/branch runs.  We won't notice that's happened until it's too late.

Reply:
> One idea, and I haven't tested this: Can you define patterns as global env values, then reference them inside !changesInclude()? (I'm guessing not, but might be worth a try).

I doesn't seem to work sadly, https://github.com/containers/podman/pull/23036/commits/83d497f91a21df77941d3e19453f8ba5417588e3

I guess Starlark is our best option for less duplication and we could split our massive yml into several smaller ones with that if this is a direction you like to go.

Reply:
Thanks for trying, shucks :_(

I cannot imagine a way to employ starlark w/o breaking `hack/get_ci_vm.sh`, so that's probably a long-term project/effort.  Though starlark could certainly make the overall CI config much easier to manage/maintain.